distributed/protocol/compression.py

```diff
@@ -143,7 +158,8 @@ def maybe_compress(payload, min_size=1e4, sample_size=1e4, nsamples=5):
     else:
         nbytes = len(payload)
 
-    if blosc and type(payload) is memoryview:
+    if blosc and type(payload) is memoryview and payload.itemsize > 1:
+        # Blosc does itemsize-aware shuffling, resulting in better compression
```

Comment:
Copying text from a longer comment earlier

> Anecdotally I've found blosc to be better when we have numeric data such as comes from numpy arrays or pandas dataframes and lz4 to be a better choice when data comes from other sources. Anecdotally I've also found that receiving a memoryview is a strong signal that our data is coming from numeric data, hence the current approach.

I still think that we should retain the old behavior.  This would come up, for example, when passing around image data (`uint8`) where I think that blosc would still be a better choice.